### ISSUE TITLE
spack install texinfo fails

### ISSUE THREAD

**Issue by vsoch** [LLNL/spack#34689]
I am getting this error:

```
# spack install
==> Concretized texinfo
 -   c64x3aa  texinfo@7.0%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
 -   6jfn4xc      ^perl@5.30.0%gcc@9.4.0~cpanm+shared+threads build_system=generic arch=linux-ubuntu20.04-x86_64

==> Installing environment /opt/env
[+] /usr (external perl-5.30.0-6jfn4xc2xmxtn6kphtjygnzc5nypwesn)
==> Installing texinfo-7.0-c64x3aaaxvyswphd6lqfrvlrpawhyk4e
==> texinfo exists in binary cache but with different hash
==> No binary for texinfo-7.0-c64x3aaaxvyswphd6lqfrvlrpawhyk4e found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/92/9261d4ee11cdf6b61895e213ffcd6b746a61a64fe38b9741a3aaa73125b35170.tar.gz
==> No patches needed for texinfo
==> texinfo: Executing phase: 'autoreconf'
==> texinfo: Executing phase: 'configure'
==> texinfo: Executing phase: 'build'
==> texinfo: Executing phase: 'install'
==> Error: InstallError: Install failed for texinfo. No such file in prefix: bin/info
See build log for details:
  /tmp/root/spack-stage/spack-stage-texinfo-7.0-c64x3aaaxvyswphd6lqfrvlrpawhyk4e/spack-build-out.txt

==> Updating view at /opt/view
==> Warning: Skipping external package: perl@5.30.0%gcc@9.4.0~cpanm+shared+threads build_system=generic arch=linux-ubuntu20.04-x86_64/6jfn4xc
```
Here is a Dockerfile to reproduce:

```dockerfile
FROM ghcr.io/rse-ops/ubuntu:20.04
RUN spack add texinfo && spack install
```
to build

```bash
$ docker build -t reproduce .
```
Thank you!

**Comment by alecbcs** [comment]
This might be related to #34312 although the error message is manifesting itself in a different way.

**Comment by vsoch** [comment]
Testing now.

**Comment by vsoch** [comment]
It installed for my test case! For the automation above, the base container is rebuilt nightly, and then it's used the next day, so I'll add a note to follow up here (and close the issue) if it's fixed. Thanks for the ping @alecbcs !

**Comment by alecbcs** [comment]
Anytime! Was following the issue this morning and saw this out of the corner of my eye when looking for another issue this afternoon!

**Comment by vsoch** [comment]
Thank you! Can we turn you into a bot? That would be the ultimate helper for a GitHub repo with this many issues! :laughing: 

> Hello, I'm alecbot! :wave:  I see that you've opened an issue that might be addressed by #34312. Have a nice day! :sun_with_face:

**Comment by vsoch** [comment]
Fixed: https://github.com/rse-ops/docker-images/actions/runs/3826674041/jobs/6510727473 thanks!
